tools/golang-json-rpc-tests/main.go

```diff
@@ -0,0 +1,412 @@
+package main
+
+import (
+    "context"
+    "crypto/ecdsa"
+    "encoding/hex"
+    "flag"
+    "fmt"
+    "log"
+    "math/big"
+    "os"
+    "strings"
+
+    "github.com/ethereum/go-ethereum"
+    "github.com/ethereum/go-ethereum/accounts/abi/bind"
+    "github.com/ethereum/go-ethereum/common"
+    "github.com/ethereum/go-ethereum/core/types"
+    "github.com/ethereum/go-ethereum/crypto"
+    "github.com/ethereum/go-ethereum/ethclient"
+    "github.com/joho/godotenv"
+)
+
+const (
+    mainnetEndpoint     = "https://mainnet.hashio.io/api"
+    testnetEndpoint     = "https://testnet.hashio.io/api"
+    previewnetEndpoint  = "https://previewnet.hashio.io/api"
+)
+
+func main() {
+    err := godotenv.Load()
+    if err != nil {
+        log.Fatalf("Error loading .env file")
+    }
+    mainnet := flag.Bool("mainnet", false, "Use mainnet network")
+    previewnet := flag.Bool("previewnet", false, "Use previewnet network")
+    privateKeyHex := os.Getenv("OPERATOR_PRIVATE_KEY")
+    flag.Parse()
+    var endpointUrl string
+    switch {
+    case *mainnet:
+        endpointUrl = mainnetEndpoint
+    case *previewnet:
+        endpointUrl = previewnetEndpoint
+    default:
+        endpointUrl = testnetEndpoint
+    }
+    client, err := ethclient.Dial(endpointUrl)
+    if err != nil {
+        log.Fatal(err)
+    }
+    fmt.Println("Connected to Ethereum client")
+    chainId, err := client.ChainID(context.Background())
+    if err != nil {
+        log.Fatalf("Failed to get chain ID: %v", err)
+    }
+    fmt.Printf("Chain ID: %s\n", chainId.String())
+    privateKeyHex = strings.TrimPrefix(privateKeyHex, "0x")
+    privateKey, err := crypto.HexToECDSA(privateKeyHex)
+    if err != nil {
+        log.Fatalf("Failed to parse private key: %v", err)
+    }
+    publicKey := privateKey.Public().(*ecdsa.PublicKey)
+    fromAddress := crypto.PubkeyToAddress(*publicKey)
+    signedTx := testSendDummyTransaction(client, fromAddress, privateKey, chainId)
+    receipt := waitForTransaction(client, signedTx)
+    blockNumber := receipt.BlockNumber
+    txIndex := receipt.TransactionIndex
+    blockHash := receipt.BlockHash
+    newestBlock := big.NewInt(0)
+    testFeeHistory(client, 5, newestBlock, []float64{10, 50, 90})
+    testBlockByNumber(client, blockNumber)
+    testTransactionByHash(client, signedTx.Hash().Hex())
+    testTransactionReceipt(client, signedTx.Hash().Hex())
+    testGetBalance(client, fromAddress)
+    testEthCall(client, fromAddress)
+    testEstimateGas(client, fromAddress)
+    testGetGasPrice(client)
+    testGetAccounts(client)
+    testBlockByHash(client, blockHash)
+    testFeeHistory(client, 5, newestBlock, []float64{10, 50, 90})
```

Comment:
duplicated with line 70?

Reply:
That's right. I've removed the duplicated line.